test/functional/shared.js

```diff
@@ -148,7 +148,7 @@ function withClient(client, callback) {
     }
     return client
       .connect()
-      .then(callback)
+      .then(callback.bind(this))
```

Comment:
do we need to make this change in `master` as well?

Reply:
yes